packages/zendesk/scripts/build.mjs

```diff
@@ -14,21 +13,48 @@ const copyFile = (fileName) => {
   fs.copySync(currPath, distPath);
 };
 
+/**
+ * Remove all the properties from manifest that are not needed for production
+ */
+const normalizeManifest = (manifest) => {
+  if (!manifest) {
+    return;
+  }
+  const manifestCopy = { ...manifest };
+  for (const key in manifestCopy.location.support) {
+    const supportApp = manifestCopy.location.support[key];
+
+    // hard coded properties for now
+    // if there are more properties in the future - prepare a smarter utility
+    delete supportApp["dev:url"];
+    delete supportApp["dev:port"];
+  }
+
+  return manifestCopy;
+};
+
+/**
+ * Prepare manifest file - if this is a local environment, serve UI files from localhost
+ * Otherwise, serve files from build folders for each respective app
+ */
 const handleManifest = (fileName) => {
   const manifest = fs.readJsonSync(fileName);
-  for (const appLocation of apps) {
-    manifest.location.support[appLocation] =
-      manifest.location.support[appLocation] || {};
-    manifest.location.support[appLocation] = {
-      ...manifest.location.support[appLocation],
+  for (const appKey in manifest.location.support) {
+    const app = manifest.location.support[appKey];
+    const devUrl = app["dev:url"] ?? "localhost";
+    const devPort = app["dev:port"] ?? "3000";
+
+    manifest.location.support[appKey] = {
+      ...manifest.location.support[appKey],
       url:
```

Comment:
This is still not working for me. If I manually add the following to the `vite.config.js`, then it works correctly:
```javascript
  server: {
    port: 3000,
  }
```

If I remove that, then vite selects a random port, even after I have added the `dev:url` and `dev:port` keys to the `manifest.json` file. Where does the build actually use the newly created `url` param?

Reply:
Okay, so the issue here is that whichever server port vite starts on should match with the `dev:port` property in the `manifest.json`. In order to have these values match up, the `server.port` should be defined in the `vite.config.js`.

So, the developer should ensure when creating a new package, to update the port (and host url) in both files. I suggest this is addressed in the `README`, under the new "Properties" section.

Reply:
Yes, that is the idea, that developer is responsible for controlling the dev server, and the boilerplate just has the ability to listen on this URl and port to serve files locally. Otherwise even if you set 'server.port' in your vite config for example to 3500, the boilerplate won't work - it will still expect to listen on 3000.
So this change is is just for this, it does not control the port in your vite config file.

But good point about README, I will update the description!
Thank you

Reply:
This also allows you to spin up multiple servers, for example, multiple frameworks, and specify which package in manifest listens to which port, so you can have sidebar app written one way, and ticket editor app maybe written in completely different way, because usually functionality is drastically different :)